lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -870,6 +947,16 @@ instance W.KeyToAddress t Seq.SeqKey => PersistState (Seq.SeqState t) where
         deleteWhere [ SeqStateAddressWalletId ==. wid ]
         deleteWhere [ SeqStateWalletId ==. wid ]
 
+    deleteCheckpointState wid sid = do
+        cpid <- selectList [ SeqStateCheckpointWalletId ==. wid
+                           , SeqStateCheckpointSlot ==. sid ] []
+        deleteWhere [ SeqStatePendingIxCheckpointId <-. fmap entityKey cpid ]
```

Comment:
Just a note that I made a mistake in the schema design, and these pending indices should not be attached to a checkpoint. But we can leave it as is now and fix the schema and query before doing the rollback function.